core/src/main/scala/chisel3/package.scala

```diff
@@ -210,26 +211,140 @@ package object chisel3 {
+     *   3. If format specifers target regular scala data types - java.lang.String.format method is called with specifier and argument to get the formatted string.
+     *
+     * Yet to document all the corener cases which this code tries to cover - but here are few of them
+     * 1. Allow calling %f or %2.2f (or something similar) on Int type data (Int, Short, Long) - this requires doing an explicit cast of the data before calling String.format
+     * 2. For bits - if no format specifier given (%s i.e) call .toPrintable to pick the default one. For regular data - let the String.format handle canonical case of %s too.
+     * Still testing this code - I am pretty sure this can be refactored and made more efficient / readable. WIP.
+     *
+     * /
+     */
+     */
+    def cf(args: Any*): Printable = {
+
+      // Handle literal % 
+      // Takes the part string - 
+      // - this is assumed to not have any format specifiers - already handled / removed before calling this function. 
+      // Only thing present is literal % if any which should ideally be with %%. 
+      // If not - then flag an error. 
+      // Return seq of Optional Printables (either PString or Percent or both - nothing else)
+      def PercentSplitter(s : String) : Seq[Option[Printable]] = {
+        if(s.isEmpty()) return Seq(Some(PString("")))
+        var iter = 0
+        var curr_start = 0
+        val buf = mutable.ListBuffer.empty[Option[Printable]]
+        while(iter < s.size) {
+          if(s(iter) == '%' ) {
+            if(iter >= s.size - 1 || s(iter + 1 ) != '%') {
+              throw new UnknownFormatConversionException("Un-escaped % found")
+            }
+            if(curr_start < iter) {
+              buf += Some(PString(s.substring(curr_start,iter)))
+            }
+            buf += Some(Percent)
+            curr_start = iter + 2
+            iter += 2
+          }
+          else {
+            iter += 1
+          }
+        }
+
+        // Handle the tail
+        if(curr_start < iter) {
+          buf += Some(PString(s.substring(curr_start,iter)))
+        }
+        buf.toSeq
       }
+
+      sc.checkLengths(args) // Enforce sc.parts.size == pargs.size + 1
       val parts = sc.parts.map(StringContext.treatEscapes)
-      // Zip sc.parts and pargs together ito flat Seq
-      // eg. Seq(sc.parts(0), pargs(0), sc.parts(1), pargs(1), ...)
+
+      // The 1st part is assumed never to contain a format specifier.
+      // If the 1st part of a string is an argument - then the 1st part will be an empty String.
+      // So we need to parse parts following the 1st one to get the format specifiers if any
+      val partsAfterFirst = parts.slice(1, parts.size)
+
+      // Align parts to their potential specifiers
+      val partsAndSpecifierSeq = partsAfterFirst.zip(args).map {
+        case (part, arg) => {
```

Comment:
```suggestion
      val partsAndSpecifierSeq = partsAfterFirst.zip(args).map { case (part, arg) =>
```
There's no need for the extra indentation and `{`